src/state/slices/tradeQuoteSlice/selectors.ts

```diff
@@ -654,6 +655,26 @@ export const selectLoadingSwappers = createSelector(
   },
 )
 
+export const selectUserValidTradeQuotes = createSelector(
+  selectTradeQuoteDisplayCache,
+  tradeQuoteDisplayCache => {
+    return tradeQuoteDisplayCache.filter(
+      quoteData =>
+        SWAPPER_USER_ERRORS.includes(quoteData.errors[0]?.error) || !quoteData.errors.length,
```

Comment:
preferably blocking: introspect all errors in the array

Reply:
Again, happy to discuss this but, we display the first error, which is the one telling the user the informations if he has the power to change anything, I don't see any values introspecting all errors, we could end up with quotes saying "Unsupported trading pair" in the top section which shouldn't be the case

Any thought? We might wan't to change the whole system and introspect every errors to display user facing errors as a priority in the interface (does this case even exists?)

Reply:
Ah yep that makes more sense. From memory, that is a limitation of our current ui and we were intending to extend it to show all errors present somehow. 

Lets say the errors on a quote are
```typescript
[
  TradeQuoteValidationError.TradingInactiveOnBuyChain, 
  TradeQuoteValidationError.UnknownError
]
```

The UI will display `TradeQuoteValidationError.TradingInactiveOnBuyChain`, and the other`UnknownError` is invisible. But should we treat it as valid in this case noting the `UnknownError` is definitely present despite being hidden?

Reply:
Seems more like a @shapeshift/product question, if we choose to put this error into the invalid quotes, users will see an error they have the power to change in the invalid quotes, seems a bit weird but legit as long as even if they fix the error they might won't be able to have a valid quote anyway

Reply:
we shouldn't display errors if they can't fix it. 
they can fix:
- switching chains or routes
- having/swapping into more assets (or gas)

an `UnknownError` example would be like reverts or something upstream right @woodenfurniture ?

Reply:
> Ah yep that makes more sense. From memory, that is a limitation of our current ui and we were intending to extend it to show all errors present somehow.
> 
> Lets say the errors on a quote are
> 
> ```ts
> [
>   TradeQuoteValidationError.TradingInactiveOnBuyChain, 
>   TradeQuoteValidationError.UnknownError
> ]
> ```
> 
> The UI will display `TradeQuoteValidationError.TradingInactiveOnBuyChain`, and the other`UnknownError` is invisible. But should we treat it as valid in this case noting the `UnknownError` is definitely present despite being hidden?

I would add that `TradingInactiveOnBuyChain` is not a shown error anyway and will end up in invalid or fully hidden in both cases

A valid example would be `SellAmountBelowTradeFee` + `Uknown Error` but I think it remains an edge case as I never faced it?